hooks/validateTokens.hook.js

```diff
@@ -0,0 +1,37 @@
+import { EntityService } from '../services/entity.service.js';
+import { TokenService } from '../services/token.service.js';
+
+export class ValidateTokens {
+  constructor(db, jwt) {
+    this.service = new EntityService(db, 'user');
+    this.tokenService = new TokenService(jwt);
+
+    this.validateRefresh = this.validateRefresh.bind(this);
+  }
+
+  async validateRefresh(request, reply) {
```

Comment:
Access token should be responsible for getting user info. Refresh token should be responsible for REFRESHING access token.